src/rgw/rgw_sal_daos.h

```diff
@@ -34,11 +34,76 @@
 #include "rgw_role.h"
 #include "rgw_sal.h"
 
+inline bool IsDebuggerAttached()
```

Comment:
This is brilliant, thank you!
I think these could live in their own header to become available to more ceph devs